torch_xla/distributed/xla_dist.py

```diff
@@ -297,34 +347,58 @@ def _run_script(script_path, client_worker):
       thread.start()
       threads.append(thread)
 
-    try:
-      for thread in threads:
-        thread.join()
-    except KeyboardInterrupt:
-      pass
-    finally:
-      self._cleanup(script_map)
-
     for thread in threads:
       thread.join()
 
-  def run(self, cmd):
+  def _run_cmd(self, cmd, script_map):
     self.logger.info(
         'Command to distribute: {}'.format(concat_cmd_list(cmd)),
         extra={
             'clientip': '',
             'ordinal': ''
         })
     self.logger.info(
-        'Cluster configuration: {}'.format(self._cluster),
+        f'Cluster configuration: {self._cluster}',
         extra={
             'clientip': '',
             'ordinal': ''
         })
-    script_map = self._prepare_scripts(cmd)
-    self._scp_scripts(script_map)
-    self._start_run(script_map)
+    try:
+      self._scp_scripts(script_map)
+      self._start_run(script_map)
+    except KeyboardInterrupt:
+      pass  # parent process already handles this
 
+  def run(self, cmd):
+    trials = RecentEvents()
```

Comment:
Oh, so that class is only for accurate tracking of failures per hour?
No need for that. Let's keep it simple and have a max-failures ceiling.

Reply:
We have been running for years workloads on Borg with only max-failures 😉

Reply:
Per discussion, done. Will add back in future if needed.